Source/FieldSolver/WarpXPushFieldsHybridPIC.cpp

```diff
@@ -0,0 +1,254 @@
+/* Copyright 2023 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald (TAE Technologies)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "Evolve/WarpXDtType.H"
+#include "FieldSolver/FiniteDifferenceSolver/FiniteDifferenceSolver.H"
+#include "FieldSolver/FiniteDifferenceSolver/HybridPICModel/HybridPICModel.H"
+#include "Particles/MultiParticleContainer.H"
+#include "Utils/TextMsg.H"
+#include "Utils/WarpXProfilerWrapper.H"
+#include "WarpX.H"
+
+using namespace amrex;
+
+void WarpX::HybridPICEvolveFields ()
+{
+    // The below deposition is hard coded for a single level simulation
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+        finest_level == 0,
+        "Ohm's law E-solve only works with a single level.");
+
+    // Perform charge deposition in component 0 of rho_fp
+    mypc->DepositCharge(rho_fp, 0._rt);
```

Comment:
I wonder if in this comment we should also make clear it that, since we pass `0._rt` as the second argument, we are depositing the charge density at the time corresponding to the current particle positions, which I guess is $n+1$ (given that the shift `-0.5_rt*dt[0]` below corresponds to $n+1/2$)?